lib/types.ts

```diff
@@ -0,0 +1,17 @@
+export type PrimitiveJsonType = 'null' | 'boolean' |
+'object' | 'array' | 'number' | 'integer' | 'string'
+
+export interface Link {
+    rel: string;
+    href: string;
+    mediaType?: string;
+}
+
+// TODO:
+// https://iot.mozilla.org/wot/#example-7-action-object
+// seems to imply input could be typed?
+export type InputType = any;
+
+export interface Subscriber {
```

Comment:
Couldn't you use the proper type here from `@types/express-ws`?

Reply:
I'd recommend using the custom type here. `express-ws` doesn't export the `WebSocket` type, so you'd need a dependency on `@types/ws`.